lib/npm-installation-manager.ts

```diff
@@ -151,15 +150,15 @@ export class NpmInstallationManager implements INpmInstallationManager {
 		return this.$npm.install(packageName, pathToSave);
 	}
 
-	private isPackageCached(packagePath: string): IFuture<boolean> {
-		return this.$fs.exists(packagePath);
-	}
-
-	private isPackageUnpacked(packagePath: string): IFuture<boolean> {
+	private isPackageUnpacked(packagePath: string, packageName: string): IFuture<boolean> {
 		return (() => {
+			let additionalDirectoryToCheck = this.packageSpecificDirectories[packageName];
 			return this.$fs.getFsStats(packagePath).wait().isDirectory() &&
-				this.$fs.exists(path.join(packagePath, "framework")).wait() &&
-				this.$fs.enumerateFilesInDirectorySync(path.join(packagePath, "framework")).length > 1;
+					(!additionalDirectoryToCheck ||
+						(this.$fs.exists(path.join(packagePath, additionalDirectoryToCheck)).wait() &&
```

Comment:
You can extract the conditions to separate method to make the code more readable:
```ts
this.hasFilesInDirectory(path.join(packagePath, additionalDirectoryToCheck))
```